Front hitos

MariaNavarro90 opened the pull request:
# Información de la Tarea
Vincular el #531 

### **Resumen de la Solución:** 
-Se hizo un nuevo maquetado sobre la seccion de hitos.

### **Información Adicional:**
-

# Descripción de los Cambios

### **Cambios:**
-

# Cómo Testear los Cambios

### **Pruebas Automáticas:**
-

### **Prubeas Manuales:**
-

# Capturas de Pantalla
<img width="2095" height="974" alt="image" src="https://github.com/user-attachments/assets/2457099e-e06d-4f52-91c7-4f26fe75be21" />

MariaNavarro90 committed:
Nueva maquetacion para la parte de hitos

MariaNavarro90 committed:
css hitos

MariaNavarro90 committed:
cambios en el boton de restaurar

MariaNavarro90 committed:
mejorar diseño del grid de hitos y ajustar alineación del encabezado

MariaNavarro90 committed:
cambiado a estilo boton poncho

MariaNavarro90 committed:
black

MariaNavarro90 committed:
djlint .

dsocial118 commented on acompanamientos/templates/acompañamiento_detail.html:
```diff
@@ -1,536 +1,1021 @@
+                                </div>
+                                <div class="hito-cell" data-label="Fecha Completado">
+                                    <span class="fecha-completado {% if not hitos.retiro_tarjeta %}vacia{% endif %}">
+                                        {{ hitos.retiro_tarjeta|yesno:'15/11/2024,—' }}
```
Bloqueante: Esta fecha hardcodeada no deberia existir. puede ser?

dsocial118 commented on acompanamientos/templates/acompañamiento_detail.html:
```diff
@@ -1,536 +1,1021 @@
-                                        {% endif %}
+                                </div>
+                            </div>
+                            <div class="hito-row {{ hitos.habilitacion_tarjeta|yesno:'completado,pendiente' }}">
```
Sugerencia: Este bloque  se repite demasiado, se podria acomodar

dsocial118 committed:
Merge branch 'development' into FrontHitos

MariaNavarro90 committed:
Comentario de PR : Bloqueante: Esta fecha hardcodeada no deberia existir. puede ser?

MariaNavarro90 committed:
Comentario PR: Sugerencia: Este bloque se repite demasiado, se podria acomodar

Código repetitivo: Eliminadas 800+ líneas duplicadas en el template
Template Tag Personalizado

  - Archivo: acompanamientos/templatetags/hitos_tags.py
  - Función: render_hito() para renderizar filas de hitos de manera
  consistente
  - Uso: {% render_hito hitos fechas_hitos campo descripcion comedor
  permisos %}
Template Parcial Reutilizable

  - Archivo: acompanamientos/templates/hitos/hito_row.html
  - Propósito: Template único para todas las filas de hitos
  - Mantiene: Estilos, lógica de estados, fechas dinámicas y permisos
Método: AcompanamientoService.obtener_fechas_hitos()
  - Funcionalidad: Extrae fechas de intervenciones que completaron cada
  hito
  - Lógica: Usa la fecha más reciente de cada hito completado
  Configuración Centralizada

  - Vista: Lista hitos_config con todos los 22 hitos
  - Beneficio: Agregar nuevos hitos es trivial (solo 1 línea)

  En resumen meti DRY

MariaNavarro90 committed:
black

MariaNavarro90 committed:
djlint
